exif/exif.go

```diff
@@ -27,6 +27,24 @@ const (
 	interopPointer = 0xA005
 )
 
+// A decodeError is returned when the image cannot be decoded as a tiff image.
+type decodeError struct {
```

Comment:
What's the reason for having this type, as opposed to just returning ErrShortReadTagValue?

(I'm guessing backwards compatibility with the error string? Fair enough if so.)